File: 57-insert-interval/insert-interval.py
class Solution(object):
    def insert(self, intervals, newInterval):
        """
        :type intervals: List[List[int]]
        :type newInterval: List[int]
        :rtype: List[List[int]]
        """

        newIntervals = []
        i = 0
        # Before finding intersection loop
        while i <len(intervals) and intervals[i][1] < newInterval[0]:
            newIntervals.append(intervals[i])
            i += 1
        
        while i<len(intervals) and intervals[i][0] <= newInterval[1]:
            newInterval = [min(intervals[i][0], newInterval[0]), 
                           max(intervals[i][1], newInterval[1])]
            i += 1

        newIntervals.append(newInterval)
        # After merging loop
        while i <len(intervals):
            newIntervals.append(intervals[i])
            i += 1

        return newIntervals
                    

File: 57-insert-interval/insert-interval.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        int i= 0;
        List<int[]> newIntervals = new ArrayList<int[]>();

        while(i<intervals.length && intervals[i][1] < newInterval[0]){
            newIntervals.add(intervals[i]);
            i++;
        }

        while(i<intervals.length && intervals[i][0] <= newInterval[1]){
            newInterval[0] = Math.min(intervals[i][0], newInterval[0]);
            newInterval[1] = Math.max(intervals[i][1], newInterval[1]);
            i++;
        }
        
        newIntervals.add(newInterval);

        while(i<intervals.length){
            newIntervals.add(intervals[i]);
            i++;
        }

        int[][] result = new int[newIntervals.size()][2];
        int currentIndex = 0;
        for(i=0; i<newIntervals.size(); i++){
            result[currentIndex] = newIntervals.get(i);
            currentIndex++;
        }
        
        return result;
    }
}